src/ir/lowering.rs

```diff
@@ -592,9 +592,12 @@ impl LowerStructDefn for StructDefn {
                 self_ty,
                 fields: fields?,
                 where_clauses,
-                flags: ir::StructFlags {
-                    external: self.flags.external,
-                    fundamental: self.flags.fundamental,
+                struct_type: match (self.flags.external, self.flags.fundamental) {
+                    // A local struct is local regardless of fundamental
+                    (false, _) => ir::StructType::Local,
```

Comment:
I think this is correct, but it was not immediately obvious to me, I have to admit. But I guess the point is that this program compiles (and clearly should), even though `Clone` is an external trait:

```rust
#![feature(fundamental)]

#[fundamental]
struct Foo<T> { t: T }

impl<T: Clone> Clone for Foo<T> {
    fn clone(&self) -> Self {
        Foo { t: self.t.clone() }
    }
}

fn main() { }
```